src/llmq/quorums_blockprocessor.cpp

```diff
@@ -299,18 +299,16 @@ bool CQuorumBlockProcessor::IsCommitmentRequired(Consensus::LLMQType llmqType, c
 }
 
 // WARNING: This method returns uint256() on the first block of the DKG interval (because the block hash is not known yet)
-uint256 CQuorumBlockProcessor::GetQuorumBlockHash(Consensus::LLMQType llmqType, const CBlockIndex* pindexPrev)
+uint256 CQuorumBlockProcessor::GetQuorumBlockHash(Consensus::LLMQType llmqType, int nHeight)
 {
     auto& params = Params().GetConsensus().llmqs.at(llmqType);
 
-    int nHeight = pindexPrev->nHeight + 1;
     int quorumStartHeight = nHeight - (nHeight % params.dkgInterval);
-    if (quorumStartHeight >= pindexPrev->nHeight) {
+    uint256 quorumBlockHash;
+    if (!GetBlockHash(quorumBlockHash, quorumStartHeight)) {
```

Comment:
We should maybe add an `AssertLockHeld(cs_main)` as otherwise the lock acquired in `GetBlockHash` might result in a surprise deadlock if we ever call this method with `minableCommitmentsCs` held